libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -500,6 +501,32 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       await doInstallInternalAsync(this.options);
     }
 
+    // if usePnpmSyncForInjectedDependencies is true
+    // the pnpm-sync will generate the pnpm-sync.json based on lockfile
+    if (this.rushConfiguration.packageManager === 'pnpm' && experiments?.usePnpmSyncForInjectedDependencies) {
+      const pnpmLockfilePath: string = this.rushConfiguration.tempShrinkwrapFilename;
+      const pnpmStorePath: string = `${this.rushConfiguration.commonTempFolder}/node_modules/.pnpm`;
+      await pnpmSyncPrepareAsync({
```

Comment:
Can this be done in a plugin too?

Reply:
Do you have examples here? I could not find examples on how to hook a plugin to `rush install` or `rush update`. Is it possible to do that?

Reply:
Let's do that in a follow-up.